src/ORM/Search/SearchContext.php

```diff
@@ -151,8 +150,38 @@ public function getQuery($searchParams, $sort = false, $limit = false, $existing
         if ($this->connective != "AND") {
             throw new Exception("SearchContext connective '$this->connective' not supported after ORM-rewrite.");
         }
+        $this->setSearchParams($searchParams);
+        $query = $this->prepareQuery($sort, $limit, $existingQuery);
+        return $this->search($query);
+    }
+
+    /**
+     * Perform a search on the passed DataList based on $this->searchParams.
+     */
+    protected function search(DataList $query): DataList
```

Comment:
```suggestion
    private function search(DataList $query): DataList
```

Reply:
Why? Someone creating a subclass of `SearchContext` might want to override the behaviour of this method without duplicating the logic in `getQuery`

Reply:
done